igvc/src/speed_compare/main.cpp

```diff
@@ -0,0 +1,83 @@
+#include <igvc_msgs/velocity_pair.h>
+#include <ros/ros.h>
+#include <std_msgs/Bool.h>
+#include <chrono>
+#include <fstream>
+
+bool enabled = false;
+std::ofstream file;
+float target_left_vel;
+float target_right_vel;
+ros::Time start_time;
+
+void enabled_callback(const std_msgs::BoolConstPtr& msg)
+{
+  enabled = msg->data;
+  if (enabled) {
+  	  ROS_INFO_STREAM("The robot is enabled. Data will be written");
+  } else {
+	  ROS_INFO_STREAM("The robot is disabled. Data will not be written.");
+  }
+}
+
+void motors_callback(const igvc_msgs::velocity_pair& msg) {
+  if (!enabled && ros::Time::now().sec <= 10000) {
```

Comment:
I think a better change would be to alter the gazebo files in order for it to publish robot enabled as always true since that is really what is happening here.